### DISCUSSION TITLE
Include audio file in executable

### DISCUSSION THREAD

**Discussion by Mistera91** [tesselode/kira#98] [Q&A] [answered]
Hello !
I am searching for a way to play some sound directly from inside the executable : for other low-level audio crates, this is, according to people online, usually done using the `include_bytes!()` macro, but in this crate, I can’t find a way to generate a `StaticSoundData` from anything else than a path.
If you manage to accomplish what I just talked about, or have any idea that could help in doing so, feel free to let me know !

**Comment by Roms1383** [comment, marked as answer]
You already can actually!

`include_bytes!()` yields a `&'static [u8; N]`, which implements `AsRef<[u8]> + 'static` and can be wrapped into a `Cursor<&[u8]>`, which both `StaticSoundData` and `StreamingSoundData` support with [from_cursor](https://docs.rs/kira/latest/kira/sound/static_sound/struct.StaticSoundData.html#method.from_cursor) :

```rs
use kira::{
    manager::{AudioManager, AudioManagerSettings, DefaultBackend},
    sound::static_sound::StaticSoundData,
};

fn main() {
    use std::io::Cursor;

    let bytes = include_bytes!("wild-cherry-play-that-funky-music.mp3");
    let cursor = Cursor::new(bytes);
    let data = StaticSoundData::from_cursor(cursor).expect("read static sound data from cursor");
    let duration = data.duration();
    let mut manager: AudioManager<DefaultBackend> =
        AudioManager::new(AudioManagerSettings::default()).expect("audio manager");
    manager.play(data).expect("play sound data");
    
    std::thread::sleep(duration);
    // depending on its size, you can experience some latency before sound gets loaded in memory and played
}
```

**Reply by Mistera91** [reply]
This works ! Thank you very much !
